invenio_app_ils/ui/src/invenio_app_ils/forms/core/VocabularyField.js

```diff
@@ -37,19 +35,11 @@ export class VocabularyField extends React.Component {
     try {
       const response = await this.query();
       const options = [];
-      if (response.data.hits.length > 0) {
-        options.push({
-          key: '',
-          value: '',
-          text: '-',
-        });
-      }
       for (let hit of response.data.hits) {
         options.push(serializer(hit));
       }
-      if (this._mounted) {
```

Comment:
@equadon why did you need this `mounted`?

Reply:
Now that accordion fields are optimized to only render if they are open it might be that you open a vocabulary field inside an accordion and if you close it before the vocabulary field has called `this.setState` you'll get an error that you can't update the state for an unmounted component.